src/config.rs

```diff
@@ -0,0 +1,69 @@
+use std::{borrow::Cow, path::PathBuf};
+
+use figment::{
+    providers::{Env, Format, Yaml},
+    Figment,
+};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(PartialEq, Debug, Serialize, Deserialize)]
+pub struct Config {
+    /// The sentry DSN to use for error reporting.
+    pub sentry_dsn: Option<String>,
+
+    /// The environment to report to sentry errors to.
+    pub sentry_env: Option<Cow<'static, str>>,
+}
+
+impl Config {
+    /// Load configuration from an optional configuration file and environment
+    pub fn extract(path: &Option<PathBuf>) -> anyhow::Result<Config> {
```

Comment:
`extract` is an odd name for this (I assume it's from the `figment` method?), I would use `load` or somesuch here.

Reply:
I'm thinking about later adding a second parameter passing the CliApp so it would extract config from both a file, or from some cli app options.

I'll leave it as extract now, I think it's ok